# Lectura de ficheros easyMoney

## import Librariesz

In [1]:
import numpy as np 
import pandas as pd
pd.options.display.float_format = '{:,.2f}'.format
import matplotlib as plt

import warnings
warnings.filterwarnings('ignore')

from ipywidgets import widgets
from ydata_profiling import ProfileReport
from ydata_profiling.utils.cache import cache_file


/Users/claudiacastro/anaconda3/lib/python3.11/site-packages/pandas/core/arrays/masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


In [2]:
# ca_df = pd.read_parquet('https://easy-money-project-bucket.s3.eu-west-3.amazonaws.com/commercial_activity_df.parquet')

# prod_df = pd.read_parquet('https://easy-money-project-bucket.s3.eu-west-3.amazonaws.com/products_df.parquet')

# sdg_df = pd.read_parquet('https://easy-money-project-bucket.s3.eu-west-3.amazonaws.com/sociodemographic_df.parquet')

In [9]:
# df_1 = pd.merge(ca_df,prod_df, how="inner" ,on=['pk_cid','pk_partition'])

# df_2 = pd.merge(df_1,sdg_df, how="inner",on=['pk_cid','pk_partition'])

# df_2.to_parquet("/Users/claudiacastro/easy_money_project/df_completo.parquet")

# carga del dataset completo
df_full = pd.read_parquet("/Users/claudiacastro/easy_money_project/df_completo.parquet")
# eliminamos columnas que no aportan información
df_full = df_full.drop(columns=['Unnamed: 0_x', 'Unnamed: 0_y', 'Unnamed: 0', "em_account_pp"], axis=1)
# eliminamos duplicados a partir de la columna pk_cid y nos quedamos con el primero
df_full = df_full.drop_duplicates(subset=['pk_cid'], keep="first")
# modificamos el tipo de dato de las columnas
df_full["pk_partition"] = pd.to_datetime(df_full["pk_partition"])
df_full["entry_date"] = pd.to_datetime(df_full["entry_date"], errors='coerce')
df_full["entry_channel"] = df_full["entry_channel"].astype('category')
df_full['active_customer'] = df_full['active_customer'].astype('int32')
df_full['segment'] = df_full['segment'].astype('category')
df_full['short_term_deposit'] = df_full['short_term_deposit'].astype('int32')
df_full['loans'] = df_full['loans'].astype('int32')
df_full['mortgage'] = df_full['mortgage'].astype('int32')
df_full['funds'] = df_full['funds'].astype('int32')
df_full['securities'] = df_full['securities'].astype('int32')
df_full['long_term_deposit'] = df_full['long_term_deposit'].astype('int32')
df_full['credit_card'] = df_full['credit_card'].astype('int32')
# df_full['payroll'] = df_full['payroll'].astype('int32')
# df_full['pension_plan'] = df_full['pension_plan'].astype('int32')
df_full['payroll_account'] = df_full['payroll_account'].astype('int32')
df_full['emc_account'] = df_full['emc_account'].astype('int32')
df_full['debit_card'] = df_full['debit_card'].astype('int32')
df_full['em_account_p'] = df_full['em_account_p'].astype('int32')
df_full['em_acount'] = df_full['em_acount'].astype('int32')
df_full["country_id"] = df_full["country_id"].astype('category')
df_full["region_code"] = df_full["region_code"].astype('category')
df_full["gender"] =  df_full["gender"].astype('category')
df_full["age"] = df_full["age"].astype('int32')
df_full["deceased"] = df_full["deceased"].astype('category')
# reporte del dataset
profile = ProfileReport(df_full, title="Profiling Report Easy Money")
profile.to_file("profiling_report_easyMoney.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [19]:
df_full.isna().mean().sort_values(ascending=False).apply(lambda x: x*100).sort_values(ascending=False).head(5)

salary          34.39
segment         29.13
entry_channel   28.89
entry_date       0.32
region_code      0.03
dtype: float64

1. La ingesta de datos se realiza en enero de 2018 y la última partición es de mayo de 2019, siendo un total 17 particiones o ingestas de datos.

2. En cada ingesta se repiten los clientes y se deciden eliminar los clientes duplicados pasando de un dataset de casi 6 millones de registro a 456,373 registros.

3. Se elimina la columna "em_account_pp" ya que ningun cliente posee una Cuenta easyMoney++

4. Las columnas con valores faltantes son:
   
    - salary  :             34.39 (%)

   - segment   :           29.13

    - entry_channel  :      28.89

    - entry_date    :        0.32

    - region_code   :        0.03

    - payroll   :            0.01

    - pension_plan  :        0.01

    - gender      :          0.01
  
5. A continuación se muestra la descripcion del dataset